src/bin/cargo-flamegraph.rs

```diff
@@ -283,8 +287,20 @@ fn main() {
         .take()
         .unwrap_or("flamegraph.svg".into());
 
+    let should_open = opt.open;
+
     flamegraph::generate_flamegraph_by_running_command(
         workload,
-        flamegraph_filename,
+        flamegraph_filename.clone(),
     );
+
+    if should_open && flamegraph_filename.exists() {
```

Comment:
`generate_flamegraph_by_running_command` exists the program when it couldnt generate the file, so it is unnecessary to check if it was created.